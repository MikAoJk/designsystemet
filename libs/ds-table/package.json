{
  "name": "@skatteetaten/ds-table",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.5.2",
  "homepage": "https://www.skatteetaten.no/stilogtone/designsystemet/",
  "repository": {
    "type": "git",
    "url": "https://github.com/Skatteetaten/designsystemet.git"
  },
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^23",
    "react": "^18",
    "react-i18next": "^13",
    "@skatteetaten/ds-core-utils": "*",
    "@skatteetaten/ds-icons": "*",
    "@skatteetaten/ds-buttons": "*"
  },
  "scripts": {}
}
