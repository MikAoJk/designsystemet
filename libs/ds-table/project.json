{
  "name": "ds-table",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-table/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.{ts,tsx,js,jsx}",
          "libs/ds-table/package.json"
        ]
      }
    },
    "build": {
      "executor": "@nx/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-table",
        "tsConfig": "libs/ds-table/tsconfig.lib.json",
        "project": "libs/ds-table/package.json",
        "entryFile": "libs/ds-table/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": ["libs/ds-table/rollup.config.js"],
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-table/README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "LICENSE",
            "input": ".",
            "output": "."
          }
        ],
        "babelUpwardRootMode": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/ds-table/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-table/**/*.css",
          "libs/ds-table/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "@skatteetaten/ds-dev-config:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-table"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish dist/libs/ds-table {args.switches}",
        "cwd": ""
      }
    }
  },
  "implicitDependencies": ["!ds-dev-config"]
}
