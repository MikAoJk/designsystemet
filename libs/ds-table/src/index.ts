export * from './Table/Table';
export * from './Table/Table.types';
