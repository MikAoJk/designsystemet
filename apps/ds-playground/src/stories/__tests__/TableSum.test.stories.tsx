import { Table } from '@skatteetaten/ds-table';
import { expect } from '@storybook/jest';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { within } from '@storybook/testing-library';

export default {
  component: Table.Sum,
  title: 'Tester/Table/Sum',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: { table: { disable: true } },
    colSpan: { table: { disable: true } },
    hasTopSeparator: { table: { disable: true } },
    labelAlignment: { table: { disable: true } },
    sumText: { table: { disable: true } },
    valueAlignment: { table: { disable: true } },
  },
} as ComponentMeta<typeof Table.Sum>;

const Template: ComponentStory<typeof Table.Sum> = (args) => (
  <div data-test-block>
    <Table.Sum {...args}>{'9 000 kr'}</Table.Sum>
  </div>
);

// Når Table har en ref, så får dom table elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ref: (instance: HTMLTableRowElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const table = canvas.getByRole('row');
  await expect(table).toBeInTheDocument();
  await expect(table).toHaveAttribute('id', 'dummyIdForwardedFromRef');
};

// Når TableSum har en id, så har table-element id
// Når TableSum har en custom CSS, så vises custom stil
// Når TableSum har en lang, så har table-element lang
// Når TableSum har dataTestid, så har table-elementet data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes (FA2-5)';
WithAttributes.args = {
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
};
WithAttributes.argTypes = {
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.parameters = {
  imageSnapshot: { disable: true },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const sum = canvas.getByTestId('123ID');
  await expect(sum).toHaveClass('dummyClassname');
  await expect(sum).toHaveAttribute('id', 'htmlId');
  await expect(sum).toHaveAttribute('lang', 'nb');
};
