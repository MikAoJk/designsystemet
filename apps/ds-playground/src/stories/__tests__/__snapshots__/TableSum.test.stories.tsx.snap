// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/Table/Sum WithAttributes play-test 1`] = `
<div data-test-block="true">
  <tfoot id="htmlId"
         class="dummyClassname"
         lang="nb"
         data-testid="123ID"
  >
    <tr class>
      <th class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  TableSum_sum__JNHMR "
          scope="row"
      >
        Sum
      </th>
      <td class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  TableSum_sum__JNHMR ">
        9 000 kr
      </td>
    </tr>
  </tfoot>
</div>
`;

exports[`Tester/Table/Sum WithRef play-test 1`] = `
<div data-test-block="true">
  <tfoot>
    <tr class
        id="dummyIdForwardedFromRef"
    >
      <th class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  TableSum_sum__JNHMR "
          scope="row"
      >
        Sum
      </th>
      <td class="TableDataCell_dataCell__fJqtw TableDataCell_dataCell_standard__ZVYHE TableDataCell_dataCell_left__+6UbM  TableSum_sum__JNHMR ">
        9 000 kr
      </td>
    </tr>
  </tfoot>
</div>
`;
